File: python-sdk/cleanster/models/property.py
"""Property model."""

from typing import Any, Dict, Optional


class Property:
    """Represents a physical property where cleanings take place."""

    def __init__(self, data: Dict[str, Any]):
        self.id: Optional[int] = data.get("id")
        self.user_id: Optional[int] = data.get("userId")
        self.name: Optional[str] = data.get("name")
        self.nickname: Optional[str] = data.get("nickName")
        self.apt: Optional[str] = data.get("apt")
        self.street: Optional[str] = data.get("street")
        self.address: Optional[str] = data.get("address")
        self.city: Optional[str] = data.get("city")
        self.state: Optional[str] = data.get("state")
        self.country: Optional[str] = data.get("country")
        self.zip_code: Optional[str] = data.get("zipCode")
        self.room_count: Optional[int] = data.get("roomCount")
        self.bathroom_count: Optional[int] = data.get("bathroomCount")
        self.service_id: Optional[int] = data.get("serviceId")
        self.is_enabled: Optional[bool] = data.get("isEnabled")
        self.is_active: Optional[bool] = data.get("isActive")
        self.is_enable: Optional[bool] = data.get("isEnable")
        self.pets: Optional[str] = data.get("pets")
        self.public_name: Optional[str] = data.get("publicName")
        self.wifi_name: Optional[str] = data.get("wifiName")
        self.wifi_password: Optional[str] = data.get("wifiPassword")
        self.laundry: Optional[bool] = data.get("laundry")
        self.garbage: Optional[str] = data.get("garbage")
        self.extra_supplies: Optional[bool] = data.get("extraSupplies")
        self.created_date: Optional[str] = data.get("createdDate")
        self.access: Optional[str] = data.get("access")
        self.supplies_location: Optional[str] = data.get("suppliesLocation")
        self.parking: Optional[str] = data.get("parking")
        self.other_note: Optional[str] = data.get("otherNote")
        self.latitude: Optional[float] = data.get("latitude")
        self.longitude: Optional[float] = data.get("longitude")
        self._raw = data

    def __repr__(self) -> str:
        return f"Property(id={self.id}, name={self.name!r}, city={self.city!r})"

File: csharp-sdk/src/Cleanster/Models/Property.cs
using System.Text.Json.Serialization;

namespace Cleanster.Models;

/// <summary>Represents a physical location where cleanings take place.</summary>
public sealed record Property
{
    [JsonPropertyName("id")]            public int    Id            { get; init; }
    [JsonPropertyName("userId")]        public int?   UserId        { get; init; }
    [JsonPropertyName("name")]          public string Name          { get; init; } = "";
    [JsonPropertyName("nickName")]      public string? NickName     { get; init; }
    [JsonPropertyName("apt")]           public string? Apt          { get; init; }
    [JsonPropertyName("street")]        public string? Street       { get; init; }
    [JsonPropertyName("address")]       public string Address       { get; init; } = "";
    [JsonPropertyName("city")]          public string City          { get; init; } = "";
    [JsonPropertyName("state")]         public string? State        { get; init; }
    [JsonPropertyName("country")]       public string Country       { get; init; } = "";
    [JsonPropertyName("zipCode")]       public string? ZipCode      { get; init; }
    [JsonPropertyName("roomCount")]     public int    RoomCount     { get; init; }
    [JsonPropertyName("bathroomCount")] public int    BathroomCount { get; init; }
    [JsonPropertyName("serviceId")]     public int    ServiceId     { get; init; }
    /// <summary>Active state - <see langword="null"/> when not returned by this endpoint.</summary>
    [JsonPropertyName("isEnabled")]     public bool?  IsEnabled     { get; init; }
    [JsonPropertyName("isActive")]      public bool?  IsActive      { get; init; }
    [JsonPropertyName("isEnable")]      public bool?  IsEnable      { get; init; }
    [JsonPropertyName("pets")]          public string? Pets         { get; init; }
    [JsonPropertyName("publicName")]    public string? PublicName   { get; init; }
    [JsonPropertyName("wifiName")]      public string? WifiName     { get; init; }
    [JsonPropertyName("wifiPassword")]  public string? WifiPassword { get; init; }
    [JsonPropertyName("laundry")]       public bool? Laundry        { get; init; }
    [JsonPropertyName("garbage")]       public string? Garbage      { get; init; }
    [JsonPropertyName("extraSupplies")] public bool? ExtraSupplies  { get; init; }
    [JsonPropertyName("createdDate")]   public string? CreatedDate  { get; init; }
    [JsonPropertyName("access")]        public string? Access       { get; init; }
    [JsonPropertyName("suppliesLocation")] public string? SuppliesLocation { get; init; }
    [JsonPropertyName("parking")]       public string? Parking      { get; init; }
    [JsonPropertyName("otherNote")]     public string? OtherNote    { get; init; }
    [JsonPropertyName("latitude")]      public double? Latitude     { get; init; }
    [JsonPropertyName("longitude")]     public double? Longitude    { get; init; }
}
